mne/io/edf/edf.py

```diff
@@ -838,48 +838,39 @@ def _read_edf_header(
                                 warn(f"Invalid patient information {key}")
 
         # Recording ID
-        meas_id = {}
         rec_info = fid.read(80).decode("latin-1").rstrip().split(" ")
-        valid_startdate = False
+        # if the measurement date is avaialble in the recording info, it's used instead
+        # of the file's meas_date since it contains all 4 digits of the year.
+        meas_date = None
         if len(rec_info) == 5:
             try:
-                startdate = datetime.strptime(rec_info[1], "%d-%b-%Y")
+                meas_date = datetime.strptime(rec_info[1], "%d-%b-%Y").replace(
+                    tzinfo=timezone.utc
+                )
+                fid.read(16)  # skip the file's meas_date
+            except Exception:
+                meas_date = None
```

Comment:
It's best to put as little stuff as possible in the `try` so it's clear where the error should occur / what's being caught
```suggestion
            except Exception:
                meas_date = None
            else:
                fid.read(16)  # skip the file's meas_date
```